CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+#######################################################################
+# The Embedded Template Library (https://www.etlcpp.com/)
+#######################################################################
+cmake_minimum_required(VERSION 3.5.0)
+project(etl)
+
+#######################################################################
+# Define an option to allow consumers of the library to 
+# specify the profile they would like to compile to.  If
+# not explicitly set CMake will attempt to choose an appropriate
+# profile based on the compiler
+#######################################################################
+set(ETL_PROFILE "DEFAULT" CACHE STRING "Defines what profile header to include.  See https://www.etlcpp.com/setup.html"
+)
+
+add_library(etl
+  src/binary.cpp
+  src/crc16_ccitt.cpp
+  src/crc16_kermit.cpp
+  src/crc32.cpp
+  src/crc64_ecma.cpp
+  src/crc8_ccitt.cpp
+  src/error_handler.cpp
+  src/pearson.cpp
+  src/random.cpp
+  src/private
+  src/private/pvoidvector.cpp
+  )
+target_include_directories(etl
+  PUBLIC
+  src/
```

Comment:
unfortunately the current layout of etl doesn't really lend itself to being added to the include path like this, although that may be something to discuss in another issue.

For example, adding `src` to the include path makes these wonderfully generic includes valid, which will almost certainly cause conflicts in enough cases for it to be an issue:
```C
#include <doxygen.h>
#include <exception.h>
#include <function.h>
#include <hash.h>
#include <jenkins.h>
#include <log.h>
#include <platform.h>
#include <utility.h>
#include <version.h>
```

I don't know how other people are currently using ETL, but I have the git repo as a submodule, and due to the layout of the etl repo ended up adding only the parent directory of the submodule to the include path in order to prevent these kinds of conflicts.

Note this isn't particularly an issue with your PR, since there isn't really a sane thing to do here currently, but it is something to think about before cmake support gets merged.

Ideally the repo would get reorganized to look something like this:

`include/etl/*.h`
`src/*.cpp`

the cmake `target_include_directories` then becomes simply `include`, with the expectation that users would consume includes like so: `#include <etl/vector.h>`

Reply:
I think that purposed layout seems reasonable.  I'm happy to include it in this PR pending @jwellbelove advice.  I don't have an Arduio to deploy on so I would not be able to personally test to see if breaking up the header location would be a problem.